common/kvstore/src/main/java/org/apache/spark/util/kvstore/InMemoryStore.java

```diff
@@ -271,9 +276,9 @@ public void put(T value) throws Exception {
       }
     }
 
-    public void delete(Object key) {
-      data.remove(asKey(key));
-      if (hasNaturalParentIndex) {
+    public Boolean delete(Object key) {
```

Comment:
At a glance, it could be primitive `boolean` here and below.

Reply:
Ah, right.